src/hooks/useNotifications.ts

```
@@ -0,0 +1,416 @@
+
+        const modalClosed = Boolean(getFromLocalStorage('notif_modal_closed'))
+        const bannerShowAtVal = getFromLocalStorage('notif_banner_show_at')
+
+        // if permission is denied, do not show the modal, rely on the reminder banner schedule
+        // the banner copy guides user to enable notifications from device settings
+        if (permissionState === 'denied') {
+            setShowPermissionModal(false)
+
+            // schedule banner if not already scheduled
+            if (!bannerShowAtVal && modalClosed) {
+                const showAt = Date.now() + (useShortDelays() ? TEN_SECONDS_MS : SEVEN_DAYS_MS)
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)